src/lib/core/CHIPError.h

```diff
@@ -123,23 +131,39 @@ class ChipError
      * @note
      *  The result is valid only if CanEncapsulate() is true.
      */
+#if __cplusplus >= 202002L
+    constexpr ChipError(Range range, ValueType value) :
+        mError(MakeInteger(range, (value & MakeMask(0, kValueLength)))) CHIP_INITIALIZE_ERROR_SOURCE(nullptr, 0, std::source_location())
```

Comment:
Won't this fail to compile if `__cplusplus >= 202002L` and `!CHIP_CONFIG_ERROR_SOURCE`?

Reply:
Good catch, fixed by the adding the loc argument to all instances of CHIP_INITIALIZE_ERROR_SOURCE